components/prize_pool/commons/prize_pool.lua

```diff
@@ -0,0 +1,352 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:PrizePool
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Class = require('Module:Class')
+local Json = require('Module:Json')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local Opponent = require('Module:Opponent') -- Note: This can be overwritten
+local Table = require('Module:Table')
+local Variables = require('Module:Variables')
+
+local WidgetInjector = Lua.import('Module:Infobox/Widget/Injector', {requireDevIfEnabled = true})
+-- local LpdbInjector = Lua.import('Module...', {requireDevIfEnabled = true})
+
+local PrizePool = Class.new(function(self, ...) self:init(...) end)
+
+local TODAY = os.date('%Y-%m-%d')
+
+-- TODO: Extract Enum handling to its own Module
+local prizeTypes = Table.map({
+	'USD', 'LOCAL_CURRENCY', 'POINTS', 'QUALIFIES', 'FREETEXT'
+}, function(value, name) return name, value end)
+
+local specialDataTypes = Table.map({
+	'LASTVS', 'GROUPSCORE', 'LASTVSSCORE'
+}, function(value, name) return name, value end)
+
+local prizeData = {
+	[prizeTypes.USD] = {
+		row = 'usdprize',
+		rowParse = function (prizePool, input, context, index)
+			prizePool.hasUSD = true
+			return prizePool:_parseInteger(input)
+		end
+	},
+	[prizeTypes.LOCAL_CURRENCY] = {
+		header = 'localcurrency',
+		headerParse = function (prizePool, input, context, index)
+			return {currency = string.upper(input)}
+		end,
+		row = 'localprize',
+		rowParse = function (prizePool, input, context, index)
+			return prizePool:_parseInteger(input)
+		end
+	},
+	[prizeTypes.QUALIFIES] = {
+		header = 'qualifies',
+		headerParse = function (prizePool, input, context, index)
+			local link = input:gsub(' ', '_')
+			local data = {link = link}
+
+			-- Automatically retrieve information from the Tournament
+			local tournamentData = prizePool:_getTournamentInfo(link)
+			if tournamentData then
+				data.title = tournamentData.tickername
+				data.icon = tournamentData.icon
+				data.iconDark = tournamentData.icondark
+			end
+
+			-- Manual override
+			local prefix = 'qualifies' .. index
+			data.title = context[prefix .. 'name'] or data.title
```

Comment:
```suggestion
			local prefix = 'qualifies' .. index
			local data = {
				link = link,
				title = context[prefix .. 'name'],
				icon = context[prefix .. 'icon'],
				iconDark = context[prefix .. 'icondark'],
			}

			-- Automatically retrieve information from the Tournament if needed
			if not (data.title and data.icon and data.iconDark) then
				local tournamentData = prizePool:_getTournamentInfo(link)
				if tournamentData then
					data.title = data.title or tournamentData.tickername
					data.icon = data.icon or tournamentData.icon
					data.iconDark = data.iconDark or tournamentData.icondark
				end
			end
```
there might be events where the page of next event does not exist yet, so for those we need to be able to add the icons etc. manually
(example from sc2 would be the DreamHackMasters which qualify the winner to the IEM already, or the DreamHackMaster Regionals where the top X are automatically qualified to the next DreamHackMaster Regional)

also in general overwrite option for the icons is imo nice
and it might be good to read first and only query if not all of them are entered (would save the query in some cases)

Reply:
I think Icons should always be retrieved from the tournament (if available), as the icons set in the prizepool manually before the tournament page is created may be a guess (eg a series icon) rather than the actual tournament icon. (and then contributors forget to remove/update from the prizepool)

Reply:
okay i see that point but imo we should still be able to add manually in case of non existing pages
you can just let it use the retrieved ones if it can retrieve and else the manually inputted ones